src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -1215,10 +1216,16 @@ public ErrorDiagnostic UnknownModuleReferenceScheme(string badScheme, ImmutableA
                 "BCP200",
                 $"{BuildInvalidOciArtifactReferenceClause(aliasName, badRef)} The registry \"{badRegistry}\" exceeds the maximum length of {maxLength} characters.");
 
-            public ErrorDiagnostic ExpectedProviderSpecification() => new(
-                TextSpan,
-                "BCP201",
-                "Expected a provider specification string of format \"<providerName>@<providerVersion>\" at this location.");
+            public ErrorDiagnostic ExpectedProviderSpecification(bool isDynamicLoadingEnabled)
+            {
+                var message = "Expected a provider specification string of format \"<providerName>@<providerVersion>\" at this location.";
```

Comment:
Isn't this syntax going to show a deprecation warning? Shouldn't we avoid recommending it in diagnostics?

Reply:
yes